metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/events/TaxonomyEventHandler.java

```diff
@@ -77,6 +77,9 @@ public Taxonomy importTaxonomyTree(Taxonomy taxonomy) throws Exception {
                 long taxIdLong = Long.parseLong(taxId);
                 Taxonomy taxonomyParent = importTaxonomyTree(new Taxonomy(taxIdLong));
                 taxonomy.setTaxonomyForRank(taxonomyParent, rankEnum.toString());
+            } else if (taxonomy.getRank().equals(RANK.SPECIES.name) && taxonomy.getTaxonomySpecies() == null) {
+                // self-reference at species level
+                taxonomy.setTaxonomySpecies(taxonomy);
```

Comment:
I think this can be simplified even more. The issue we're solving here affects not only species taxonomies; if we import an order taxonomy, its `taxonomyOrder` field will also be empty because of the same problem.

Instead, you could include a check inside the for-loop: if the `rank` is the same as `rankEnum` (meaning the taxonomy's rank is one of the fixed-rank list), then call `setTaxonomyForRank` with the `taxonomy` itself.